src/test/groovy/org/gradle/android/ConfigurationCachingTest.groovy

```diff
@@ -22,7 +22,10 @@ class ConfigurationCachingTest extends AbstractTest {
                 .build()
 
         then:
-        !result.output.contains("problems were found storing the configuration cache")
+        // In Gradle 7.4 the build always uses the following words.
+        if (!result.output.contains("0 problems were found storing the configuration cache")) {
+            !result.output.contains("problems were found storing the configuration cache")
+        }
```

Comment:
I find this conditional hard to reason about.  It might be better to use a method so that the intention is really clear - something like:
```
static final String CC_PROBLEMS_FOUND = "problems were found storing the configuration cache"

...

then:
assertNoConfigCacheProblemsFound(result)

...

void assertNoConfigCacheProblemsFound(BuildResult result) {
    assert !result.output.contains(CC_PROBLEMS_FOUND) || result.output.contains("0 ${CC_PROBLEMS_FOUND}")
}
```

Reply:
This is a good point. Thanks for the clear code snippet!